src/encoded/static/components/navigation/components/SearchBar.js

```diff
@@ -175,16 +177,32 @@ export class SearchBar extends React.PureComponent {
         const { href, currentAction } = this.props;
         const lastQuery = searchFilters.searchQueryStringFromHref(href);
 
-        this.setState(function({ typedSearchQuery: currQueryStr }){
-            let typedSearchQuery = currQueryStr; // No change - default
-            if (SearchBar.hasInput(lastQuery) && !SearchBar.hasInput(currQueryStr)) {
-                // Replace new value entered with current search query in URL if new value is empty string
-                typedSearchQuery = lastQuery;
+        setTimeout(function () {
+            this.setState(function ({ typedSearchQuery: currQueryStr, isSearchItemTypeDropDownOpen }) {
+                let typedSearchQuery = currQueryStr; // No change - default
+                if (SearchBar.hasInput(lastQuery) && !SearchBar.hasInput(currQueryStr)) {
+                    // Replace new value entered with current search query in URL if new value is empty string
+                    typedSearchQuery = lastQuery;
+                }
+                // Prevent closing onBlur if on selection view, or if have input.
+                const isVisible = isSelectAction(currentAction) || SearchBar.hasInput(typedSearchQuery) || isSearchItemTypeDropDownOpen || false;
+                return { typedSearchQuery, isVisible, isSearchItemTypeDropDownOpen: isVisible ? isSearchItemTypeDropDownOpen : false };
+            });
+        }.bind(this), 100);
+    }
+
+    onToggleSearchItemType(isOpen, event) {
+        this.setState(function (state, props) {
+            return { isSearchItemTypeDropDownOpen: isOpen };
+        }, function () {
+            const { typedSearchQuery, isSearchItemTypeDropDownOpen } = this.state;
```

Comment:
We could probably remove `isVisible` from state at least if no longer being used (?). idk.

Reply:
@alexkb0009 isVisible is still being used for search bar visibility. But isSearchItemTypeDropDownOpen added for dropdown toggle, since search bar get hidden when text input lost focus while user clicking dropdown.